tests/invoices_test.rs

```diff
@@ -0,0 +1,236 @@
+    let pdf_data = load_test_file("test.pdf");
+    let request = create_invoice_request_with_files(
+        &invoice,
+        vec![("test.jpg", jpg_data), ("doc.pdf", pdf_data)],
+    );
+
+    let response = app.oneshot(request).await.unwrap();
+
+    assert_eq!(response.status(), StatusCode::CREATED);
+}
+
+#[tokio::test]
+async fn create_invoice_with_multiple_rows_succeeds() {
+    let app = create_test_app().await;
+    let invoice = invoice_with_multiple_rows();
+    let request = create_invoice_request(&invoice);
+
+    let response = app.oneshot(request).await.unwrap();
+
+    assert_eq!(response.status(), StatusCode::CREATED);
+}
+
+#[tokio::test]
+async fn reject_invalid_iban() {
+    let app = create_test_app().await;
+    let invoice = invoice_with_invalid_iban();
+    let request = create_invoice_request(&invoice);
+
+    let response = app.oneshot(request).await.unwrap();
+
+    assert_eq!(response.status(), StatusCode::UNPROCESSABLE_ENTITY);
+}
+
+#[tokio::test]
+async fn reject_invalid_phone_number() {
+    let app = create_test_app().await;
+    let invoice = invoice_with_invalid_phone();
+    let request = create_invoice_request(&invoice);
+
+    let response = app.oneshot(request).await.unwrap();
+
+    assert_eq!(response.status(), StatusCode::UNPROCESSABLE_ENTITY);
+}
+
+#[tokio::test]
+async fn reject_empty_rows() {
+    let app = create_test_app().await;
+    let invoice = invoice_with_empty_rows();
+    let request = create_invoice_request(&invoice);
+
+    let response = app.oneshot(request).await.unwrap();
+
+    assert_eq!(response.status(), StatusCode::UNPROCESSABLE_ENTITY);
+}
+
+#[tokio::test]
+async fn reject_negative_unit_price() {
+    let app = create_test_app().await;
+    let invoice = invoice_with_negative_price();
+    let request = create_invoice_request(&invoice);
+
+    let response = app.oneshot(request).await.unwrap();
+
+    assert_eq!(response.status(), StatusCode::UNPROCESSABLE_ENTITY);
+}
+
+#[tokio::test]
+async fn reject_zero_unit_price() {
+    let app = create_test_app().await;
+    let invoice = invoice_with_zero_price();
+    let request = create_invoice_request(&invoice);
+
+    let response = app.oneshot(request).await.unwrap();
+
+    assert_eq!(response.status(), StatusCode::UNPROCESSABLE_ENTITY);
+}
+
+#[tokio::test]
+async fn reject_subject_exceeding_max_length() {
+    let app = create_test_app().await;
+    let invoice = invoice_with_long_subject();
+    let request = create_invoice_request(&invoice);
+
+    let response = app.oneshot(request).await.unwrap();
+
+    assert_eq!(response.status(), StatusCode::UNPROCESSABLE_ENTITY);
+}
+
+#[tokio::test]
+async fn reject_empty_subject() {
+    let app = create_test_app().await;
+    let invoice = invoice_with_empty_subject();
+    let request = create_invoice_request(&invoice);
+
+    let response = app.oneshot(request).await.unwrap();
+
+    assert_eq!(response.status(), StatusCode::UNPROCESSABLE_ENTITY);
```

Comment:
Could we test that the error format is correct here, since that has caused problems previously